helper/expr_string.ts

```diff
@@ -120,11 +121,9 @@ const isInstanceOfString = isInstanceOf(String);
  * console.log(isExprString("foo")); // outputs: false
  * ```
  */
-export function isExprString(x: unknown): x is ExprString {
-  return isObjectOf({
-    [EXPR_STRING_MARK]: isLiteralOf(1),
-  })(x);
-}
+export const isExprString: Predicate<ExprString> = isObjectOf({
+  [EXPR_STRING_MARK]: isLiteralOf(1),
+}) as unknown as Predicate<ExprString>;
```

Comment:
I think we can write it like

```ts
export const isExprString: Predicate<ExprString> = isIntersectionOf([
  isString,
  isObjectOf({[EXPR_STRING_MARK]: isLiteralOf(1)}),
]);
```

Reply:
Unfortunately, it is defined as a fake `string` primitive whose real type is a `String` object.
https://github.com/vim-denops/deno-denops-std/blob/b2e994301e102d75ff6af53e945f975bd295cc58/helper/expr_string.ts#L86-L87

This is done for convenience so that it can be passed to functions that expect a string.

Also, this is a mechanism for testing objects unknown to developers, not objects unknown to the std library - in other words, it is expected that `[EXPR_STRING_MARK]` properties do not come from anywhere else std library.

Reply:
I see. Please comment that to code so we can remember the context in the future 👍